pkg/ns/ns.go

```diff
@@ -74,11 +75,22 @@ func GetCurrentNS() (NetNS, error) {
 
 // Returns an object representing the namespace referred to by @path
 func GetNS(nspath string) (NetNS, error) {
+	const NSFS_MAGIC = 0x6e736673
+
 	fd, err := os.Open(nspath)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("Failed to open %v: %v", nspath, err)
+	}
+
+	stat := syscall.Statfs_t{}
+	if err = syscall.Statfs(nspath, &stat); err != nil {
+		return nil, fmt.Errorf("Failed to Statfs %v: %v", nspath, err)
 	}
-	return &netNS{file: fd}, nil
+	if stat.Type != NSFS_MAGIC {
+		return nil, fmt.Errorf("no network namespace mounted on file %q", nspath)
+	}
+
+	return &netNS{file: fd, mounted: true}, nil
```

Comment:
The Travis tests are failing because of this mounted:true.  Per my other comments, I don't think we should set mounted:true for namespaces that pkg/ns doesn't itself create, since that will attempt to unmount the namespace which is going to fail if other holders have it open.  Only the creator of the namespace should try to unmount it on Close().